uPy/html.py

```diff
@@ -0,0 +1,191 @@
+# Simple HTML parser for PMT to use
+# Curran Hyde
+#
+#   Returns a list of dictionaries. Each dictionary is a form:
+#       
+#       Pick a form by indexing into the list. Each form is a dictionary.
+#      
+#       Dictionary key value pairs are keys and values within that tag line:
+#            ex from html: name="formname" method="POST"
+#            becomes:   dict["name"] -> "formname"
+#                       dict["method"] -> "POST"
+#   Structure of return value:
+#       return value will be a list
+
+def parse_forms(html):
+    currentForm = -1
+    forms = []
+    formOpen = 0 #indicates if <form ...> seen but </form> not seen yet
+
+    #int to str
+    html = html.decode('utf-8')
+
+    # empty condition: give back empty dict
+    if(html == ''):
+        return []
+    
+    # Note: Tags can end on same line they were created on
+
+    # State List
+    # NOT_IN_TAG = 0  # 0 = not in a tag. b'<' not seen yet
+    # IN_TAG = 1      # 1 = inside a tag. b'<' seen but b'>' not seen
+
+    state = 0
```

Comment:
change this to "in_tag = false" and delete state altogether as well as comments above